### PR TITLE
Fake integration

### PR DESCRIPTION
**Pull request by erikzaadi**
# Description

Fake integration

- [x] Ensure that docker image is built, but not added to the offical `integrations.json`

## Type of change

Please leave one option from the following and delete the rest:

- [x] New feature (non-breaking change which adds functionality)
- [x] New Integration (non-breaking change which adds a new integration)

### Core testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync finishes successfully
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected

### Preflight checklist

- [x] Implemented the code in async

image

### CODE REVIEW

File: integrations/fake-integration/.port/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      - kind: fake-person
+      - kind: fake-department
+      # - kind: <ResourceName2>
+# configurations:
+#   - name: mytest-integrationToken
+#     # required: true
+#     type: string
+#     sensitive: true
+#   - name: someApplicationUrl
+#     type: url
```
Comment (Tankilevitch) [inline integrations/fake-integration/.port/spec.yaml:16]: should remove?

File: integrations/fake-integration/.port/resources/port-app-config.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      query: 'true'
+    port:
+      entity:
+        mappings:
+          identifier: .id
+          title: .name
+          blueprint: '"fake-department"'
+          properties:
+            name: .name
+            id: .id
```
Comment (Tankilevitch) [inline integrations/fake-integration/.port/resources/port-app-config.yml:30]: order should be the opposite? as that the order in which we sync is ordered by the list, so you are trying to map fake user to department that wasn't created yet

File: integrations/fake-integration/.env.example [hunk 3]
```diff
@@ -0,0 +1,2 @@
+OCEAN__PORT__CLIENT_ID="<port-client-id>"
+OCEAN__PORT__CLIENT_SECRET="<port-client-secret>"
```
Comment (Tankilevitch) [inline integrations/fake-integration/.env.example:2]: add `base_url` as well?

File: integrations/fake-integration/.port/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+description: Fake integration for Port Ocean
+icon: Cookiecutter  # Should be one of the available icons in Port
+features:
+  - type: exporter
+    section: Under Development  # Should be one of the available sections in Port
```
Comment (Tankilevitch) [inline integrations/fake-integration/.port/spec.yaml:5]: should remove the comment?

Comment (Tankilevitch) [reply integrations/fake-integration/.port/spec.yaml:5]: or adjust to why you choose this one 🤔

File: integrations/fake-integration/CONTRIBUTING.md
Comment (Tankilevitch) [inline integrations/fake-integration/CONTRIBUTING.md:1]: I feel like the contributing here should actually state what is this kind of integration and how to use it, as its a fake integration as it contains sanity and examples tests

File: integrations/fake-integration/main.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+
+@ocean.on_start()
+async def on_start() -> None:
+    print("Starting fake integration!")
+
+
+@ocean.router.get("/department/{department_id}/employees/{limit}")
+def get_employees_per_department(department_id: str, limit: int) -> Dict[str, Any]:
+    return generate_fake_persons(department_id, limit)
```
Comment (Tankilevitch) [inline integrations/fake-integration/main.py:38]: I think this should be in another file, some kind of `fake_integration_app.py`

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Nice job, added some comments

File: integrations/fake-integration/.env.example (hunk 3 above)
Comment (erikzaadi) [reply integrations/fake-integration/.env.example:2]: Interesting, some of our integrations have it in the `.env.example`, others don't.

File: integrations/fake-integration/tests/test_sync.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        },
+    )
+
+    results = await get_raw_result_on_integration_sync_kinds(INTEGRATION_PATH)
+
+    assert len(results) > 0
+
+    assert "fake-person" in results
+    assert "fake-department" in results
+
```
Comment (Tankilevitch) [inline integrations/fake-integration/tests/test_sync.py:47]: I wonder what if I have two mappings for the same kind, e.g. in jira I can have one selector for one JQL and another one for other JQL, how will I be able to differentiate between their raw results?

Comment (erikzaadi) [reply integrations/fake-integration/tests/test_sync.py:47]: Most def, the results should differ according to the selector, you can see examples in aws

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**